app/models/membership_search.rb

```diff
@@ -1,51 +1,70 @@
 # frozen_string_literal: true
 
 # MembershipSearch provides the `search`` extension method
-# for Roles.membership_dataset to allow text searching
+# for Role membership datasets to allow text searching
 # and paging of Role members
-module MembershipSearch
-  # @param search [String] - a search term in the resource id
-  def search(search: nil, kind: nil)
-    filter_kind(kind).textsearch(search)
-  end
-
-  def filter_kind(kind)
-    return self unless kind
+class MembershipSearch < Module
 
-    where(Sequel.lit("kind(member_id) in ?", Array(kind))) 
+  def initialize(id_column)
```

Comment:
I don't think you need to jump through the hoops of defining new methods here. The methods in the Module you pass to `extend:` have access to information about the association, as described here: https://guides.rubyonrails.org/v4.2/association_basics.html#association-extensions .

Also, the method has been renamed `extending` in Rails 4 and later, so you should probably update the associations to use the new name.